EssentialApp/EssentialAppTests/FeedUIIntegrationTests.swift

```diff
@@ -317,11 +318,28 @@ final class FeedUIIntegrationTests: XCTestCase {
 		wait(for: [exp], timeout: 1.0)
 	}
 	
+	func test_didSelectHandler_getsCalledWithSelectedFeedImage() {
+		let image0 = makeImage()
+		let image1 = makeImage()
+
+		var selectedImages = [FeedImage]()
+		let (sut, loader) = makeSUT(didSelectHandler: { image in
+			selectedImages.append(image)
+		})
+		sut.loadViewIfNeeded()
+		loader.completeFeedLoading(with: [image0, image1], at: 0)
+
+		sut.simulateFeedImageSelection(at: 1)
+		sut.simulateFeedImageSelection(at: 0)
+
+		XCTAssertEqual(selectedImages, [image1, image0])
```

Comment:
Missing assertion for the pre-condition:

```suggestion
		loader.completeFeedLoading(with: [image0, image1], at: 0)
		XCTAssertEqual(selectedImages, [])
		
		sut.simulateFeedImageSelection(at: 1)
		XCTAssertEqual(selectedImages, [image1])
		
		sut.simulateFeedImageSelection(at: 0)
		XCTAssertEqual(selectedImages, [image1, image0])
```

Reply:
Done